table/table.go

```diff
@@ -343,6 +344,64 @@ func OpenInMemoryTable(data []byte, id uint64, opt *Options) (*Table, error) {
 }
 
 func (t *Table) initBiggestAndSmallest() error {
+	// This defer will help gathering debugging info incase initIndex crashes.
+	defer func() {
+		if r := recover(); r != nil {
+			debug.PrintStack()
+			// Use defer for printing info because there may be an intermediate panic.
+			var debugBuf bytes.Buffer
+			defer func() {
+				fmt.Printf("%s\n== Recovered ==\n", debugBuf.String())
+				os.Exit(1)
+			}()
+
+			// Get the count of null bytes at the end of file. This is to make sure if there was an
+			// issue with mmap sync or file copy.
+			count := 0
+			for i := len(t.Data) - 1; i >= 0; i-- {
+				if t.Data[i] != 0 {
+					break
+				}
+				count++
+			}
+
+			fmt.Fprintf(&debugBuf, "\n== Recovering from initIndex crash ==\n")
+			fmt.Fprintf(&debugBuf, "File Info: [ID: %d, Size: %d, Zeros: %d]\n",
+				t.id, t.tableSize, count)
+
+			if t.shouldDecrypt() {
+				fmt.Fprintf(&debugBuf, "dataKey: %+v ", t.opt.DataKey)
```

Comment:
Do you need this? This DataKey contains information about encryption.

Reply:
Thanks. We should not log the encryption key. Addressed.